pyblazing/pyblazing/apiv2/context.py

```diff
@@ -830,6 +834,10 @@ def __init__(self, dask_client=None, network_interface=None, allocator="managed"
         for option in config_options:
             self.config_options[option.encode()] = str(config_options[option]).encode() # make sure all options are encoded strings
 
+        host_memory_quota = 0.95
```

Comment:
I would set the default to be lower than 0.95. I think 0.75 would be better. 
Also, I think we should assume that if a user is manually setting that, that they know that each worker will be using effectively the same pool of memory since its the same resource. We can help with that by adding a little more explanation in the comments about BLAZING_HOST_MEM_RESOURCE_CONSUMPTION_THRESHOLD.
Also, i think that the default should always be adjusted based on the number of GPUs per server. You have logic that already tries to do that, but I think it should be easy to extrapolate to even if its not just one machine. Assuming a default of 0.75, you can basically do:
0.75 * len(set([value['host'] for key,value in self.dask_client.scheduler_info()["workers"].items()])) /  len(self.dask_client.scheduler_info()["workers"])
Its not perfect, but in most situations you will have the same number of GPUs per server.

So in summary, lets make it so that if the user sets something, you leave it as it is. If the user does not provide a value, then set a value depending on the number of gpus per server

Reply:
Fixed. Please, check again.